oprf/group/suite.go

```diff
@@ -0,0 +1,251 @@
+package group
+
+import (
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/binary"
+	"errors"
+	"hash"
+	"io"
+	"math/big"
+
+	"github.com/armfazh/h2c-go-ref"
+)
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+// Ciphersuite object determines the prime-order group (pog) that is used for
+// performing the OPRF operations, along with the different hash function
+// definitions.
+// Should be created using NewSuite, using an string.
+type Ciphersuite struct {
+	// name of the ciphersuite.
+	name string
+
+	// dst is a tag to be used for hashing to curve.
+	dst []byte
+
+	// hash defines the hash function to be used.
+	hash hash.Hash
+
+	curve elliptic.Curve
+}
+
+// Suite is an interface that defines operations within additive
+// groups of prime order. This is the setting in which the OPRF operations
+// take place.
+type Suite interface {
```

Comment:
Why do we call this `Suite` instead of, say, `Group`? Most of the functions on this interface seem to apply to the group. It seems like a `Suite` should generically be an interface similar to something like so:

```
type Group interface {
   // All the group-related functions from below!
}

type Suite interface {
   Name() string
   Identifier() uint16
   Hash() hash.Hash
   Group() Group
}
```

Reply:
Mmm.. yes, that looks good. But at some point, I've been thinking of why we need the interfaces themselves. I doesn't seem as anyone else will instantiate them with a different object than `CipherSuite`.

Reply:
I'm not sure I followed :-\

Reply:
I removed the interface all together.